launcher/ui/WinDarkmode.cpp

```diff
@@ -4,28 +4,90 @@
+}
 
-    HMODULE hUxtheme = LoadLibraryExW(L"uxtheme.dll", NULL, LOAD_LIBRARY_SEARCH_SYSTEM32);
-    HMODULE hUser32 = GetModuleHandleW(L"user32.dll");
+BOOL IsWindows10_Only()
+{
+    HMODULE hKern32 = GetModuleHandleW(L"kernel32.dll");
+    HMODULE hNtuser = GetModuleHandleW(L"ntdll.dll");
+    return GetProcAddress(hKern32, "SetThreadSelectedCpuSets") != NULL
+            && GetProcAddress(hNtuser, "ZwSetInformationCpuPartition") == NULL; 
+}
+
+BOOL IsWindows8_0_Only()
+{
+    HMODULE hKern32 = GetModuleHandleW(L"kernel32.dll");
+    return GetProcAddress(hKern32, "CreateFile2") != NULL //  Export added in 6.2 (8)
+        && GetProcAddress(hKern32, "AppXFreeMemory") != NULL;  // Export added in 6.2 (8), removed in 6.3 (8.1)
+}
 
-    // For those confused how this works, dlls have export numbers but some can have no name and these have no name. So an address is gotten by export number. If this ever changes, it will break.
-    fnAllowDarkModeForWindow AllowDarkModeForWindow = (fnAllowDarkModeForWindow)(PVOID)GetProcAddress(hUxtheme, MAKEINTRESOURCEA(133));
-    fnSetPreferredAppMode SetPreferredAppMode = (fnSetPreferredAppMode)(PVOID)GetProcAddress(hUxtheme, MAKEINTRESOURCEA(135));
-    fnSetWindowCompositionAttribute SetWindowCompositionAttribute = (fnSetWindowCompositionAttribute)(PVOID)GetProcAddress(hUser32, "SetWindowCompositionAttribute");
+BOOL IsWindows8_1_Only()
+{
+    HMODULE hKern32 = GetModuleHandleW(L"kernel32.dll");
+    return GetProcAddress(hKern32, "CalloutOnFiberStack") != NULL //  Export added in 6.3 (8.1), Removed in 10.0.10586
+        && GetProcAddress(hKern32, "SetThreadSelectedCpuSets") == NULL; // Export added in 10.0 (10)
+}
 
-    SetPreferredAppMode(AllowDark);
-    AllowDarkModeForWindow(hwnd, dark);
+void setWindowDarkModeEnabled(HWND hWnd, bool Enabled)
+{
+    AllowDarkModeForWindow(hWnd, Enabled);
+    BOOL DarkEnabled = (BOOL)Enabled;
     WINDOWCOMPOSITIONATTRIBDATA data = {
         WCA_USEDARKMODECOLORS,
-        &dark,
-        sizeof(dark)
+        &DarkEnabled,
+        sizeof(DarkEnabled)
     };
-    SetWindowCompositionAttribute(hwnd, &data);
+
+    constexpr int NtUserSetWindowCompositionAttribute_NT6_2 = 0x13b4;
+    constexpr int NtUserSetWindowCompositionAttribute_NT6_3 = 0x13e5;
+    if (IsWindows8_0_Only())
+        WinSyscall<NtUserSetWindowCompositionAttribute_NT6_2>(hWnd, &data);
+    if (IsWindows8_1_Only())
+        WinSyscall<NtUserSetWindowCompositionAttribute_NT6_3>(hWnd, &data);
+    if (IsWindows10_Only() || IsWindows11())
+    {
+        ((fnSetWindowCompositionAttribute)(PVOID)GetProcAddress(GetModuleHandleW(L"user32.dll"), "SetWindowCompositionAttribute"))
+            (hWnd, &data);
+        // Verified this ordinal is the same through Win11 22H2 (5/8/2023)
+        ((fnSetPreferredAppMode)(PVOID)GetProcAddress(GetModuleHandleW(L"uxtheme.dll"), MAKEINTRESOURCEA(135)))
+            (AppMode_AllowDark);
+    }
```

Comment:
If these are mutually exclusive wouldn't it make more sense to use `else if` instead of `if` for the subsequent checks